### PR TITLE
Adds unknown account and container.

### PR DESCRIPTION
**Pull request by xiahome**
This PR adds constants for unknown account and container. They are
specifically defined for the blobs put without specifying target account
and container.

./gradlew build && ./gradlew test

### CODE REVIEW

File: ambry-api/src/main/java/com.github.ambry/account/Account.java [hunk 1]
```diff
@@ -72,8 +72,30 @@
...
-  // accountId == UNKNOWN_ACCOUNT_ID indicate accountId is not available at the time when this blobId is formed.
+  /**
+   * The id of {@link #UNKNOWN_ACCOUNT}.
+   */
   public static final short UNKNOWN_ACCOUNT_ID = -1;
+
+  /**
+   * The name of {@link #UNKNOWN_ACCOUNT}.
+   */
+  public static final String UNKNOWN_ACCOUNT_NAME = "ambry-unknown-account-name";
```
Comment (xiahome) [inline ambry-api/src/main/java/com.github.ambry/account/Account.java]: Removed `x-`

**Comment by xiahome** [comment]
Re @cgtz comment: I would prefer to have the `UNKNOWN_ACCOUNT` onboarded in the same way as other accounts, so zk will be the single source of truth for all the accounts, and `HelixAccountService` does not need to have special logic to handle conflict for this account.

File: ambry-api/src/main/java/com.github.ambry/account/Container.java [hunk 2]
```diff
@@ -50,8 +50,47 @@
...
+  public static final ContainerStatus UNKNOWN_CONTAINER_STATUS = ContainerStatus.ACTIVE;
+
+  /**
+   * The description of {@link #UNKNOWN_CONTAINER}.
+   */
+  public static final String UNKNOWN_CONTAINER_DESCRIPTION =
+      "This is a container for the blobs without specifying a target account and container when they are put";
+
+  /**
+   * The privacy setting of {@link #UNKNOWN_CONTAINER}. {@code true} means it will delegate to the actual ACL
```
Comment (vgkholla) [inline ambry-api/src/main/java/com.github.ambry/account/Container.java]: I don't think you need to describe what's done if this is `true`/`false` here.

File: ambry-api/src/test/java/com.github.ambry/account/AccountContainerTest.java [hunk 3]
```diff
@@ -468,6 +468,31 @@ public void testRemoveNonExistContainer() throws JSONException {
   }
 
   /**
+   * Tests for {@link Account#UNKNOWN_ACCOUNT} and {@link Container#UNKNOWN_CONTAINER}.
+   */
+  @Test
+  public void testUnknownAccountAndContainer() {
```
Comment (vgkholla) [inline ambry-api/src/test/java/com.github.ambry/account/AccountContainerTest.java]: do we need a test for this? My vote would be for no - this is just java object construction - no functionality.

Comment (xiahome) [inline ambry-api/src/test/java/com.github.ambry/account/AccountContainerTest.java]: This is to ensure all the internal values are as expected.

File: ambry-api/src/main/java/com.github.ambry/account/Container.java (hunk 2 above)
Comment (xiahome) [inline ambry-api/src/main/java/com.github.ambry/account/Container.java]: Removed.

**Pull request merged**